### DISCUSSION TITLE
[CONTRACTS] - Reviving uAD on swap rewards in UBQ

### DISCUSSION THREAD

**Discussion by 0x4007** [pavlovcik/uad-ui#200] [Ideas]
@zgorizzo69 do you think its realistic to automatically add/subtract from your current pending bonding rewards?

![image](https://user-images.githubusercontent.com/4975670/134466329-98cd78f3-0cb7-4a6b-bf1f-a55c9d0bc34c.png)

### Example

You buy uAD when price below $1.00 and you get a small boost to your pending bonding rewards (more UBQ is withdraw-able) and if you sell immediately afterwards, then we slash some of your pending bonding rewards? This way we can "undo" somebody's reward if they undo the help to the system.

### Exploit Vector

User buys uAD below $1.00, withdraws pending UBQ rewards, then dumps uAD while its still below $1.00. IF we can go negative in the pending emissions then we're covered, assuming that we can not go negative then this is an exploit vector. 

Thoughts? @hpiyankov @zgorizzo69

**Comment by zgorizzo69** [comment]
That seems pretty hard to do there is no relation between uAD and your bonding share.the only way would be to add that logic in the transfer hook but then what happen if the user has two or more bonding shares ?

**Comment by 0x4007** [comment]
The relation would be your wallet address, and if it has any existing bonding shares with pending rewards. Assuming that we can do a negative pending reward balance I do not think that there is a functional difference between any. We can simply modify the bonding share with the lowest ID!

**Comment by hpiyankov** [comment]
I don't see an economic issue with this, but also I feel that this is more in the venue of "quality of life improvements" rather than something which will be a core stabilisation mechanic. 

Right now we are lacking demand and this is why the price is sliding, I don't think that if we add this feature, it will improve on the situaiton.
